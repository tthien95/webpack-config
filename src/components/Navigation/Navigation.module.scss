$content-width: 1000px;
$nav-height: 70px;
$nav-background: #fff;
$nav-font-color: #0c0c0c;
$link-hover-color: #f8b500;
$breakpoint: 799px;

header {
  height: $nav-height;
  background: $nav-background;
}

.wrap {
  max-width: $content-width;
  margin: 0 auto;
}

.logo {
  position: absolute;
  padding-left: 20px;
  line-height: $nav-height;
  text-transform: uppercase;
  font-size: 1.4em;

  a,
  a:visited {
    color: $nav-font-color;
    text-decoration: none;
  }
}

nav {
  float: right;
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
    li {
      float: left;
      position: relative;
      a {
        display: block;
        padding: 10px 20px 0 20px;
        line-height: 50px;
        background: $nav-background;
        color: $nav-font-color;
        text-decoration: none;

        &::before {
          transition: 300ms;
          height: 5px;
          content: '';
          position: absolute;
          background-color: $link-hover-color;
          width: 0%;
          bottom: 0px;
        }

        &.nav-active::before,
        &:hover::before {
          width: 70%;
        }
      }
    }
  }
}

.nav-mobile {
  display: none;
  position: absolute;
  top: 0;
  right: 0;
  background: $nav-background;
  height: $nav-height;
  width: $nav-height;
}

#nav-toggle {
  position: absolute;
  left: 18px;
  top: 22px;
  cursor: pointer;
  padding: 10px 35px 16px 0px;
  span,
  span:before,
  span:after {
    cursor: pointer;
    border-radius: 1px;
    height: 5px;
    width: 35px;
    background: $nav-font-color;
    position: absolute;
    display: block;
    content: '';
    transition: all 300ms ease-in-out;
  }
  span:before {
    top: -10px;
  }
  span:after {
    bottom: -10px;
  }
  &.active span {
    background-color: transparent;
    &:before,
    &:after {
      top: 0;
    }
    &:before {
      transform: rotate(45deg);
    }
    &:after {
      transform: rotate(-45deg);
    }
  }
}

@media only screen and (max-width: $breakpoint) {
  .nav-mobile {
    display: block;
  }
  nav {
    width: 100%;
    max-height: 0;
    overflow: hidden;
    padding: $nav-height 0 15px;
    transition: max-height 0.4s ease-in-out;

    ul li {
      float: none;
      a {
        padding: 15px;
        line-height: 20px;

        &.nav-active::before,
        &:hover::before {
          width: 20%;
        }
      }
    }

    &.expand {
      max-height: 20em;
    }
  }
}
