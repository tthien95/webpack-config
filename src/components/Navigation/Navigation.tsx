import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navigation.module.scss';

const activeStyle = ({ isActive }: { isActive: boolean }) =>
  isActive ? styles['nav-active'] : undefined;

const menuButtonHandler = (event: React.MouseEvent) => {
  event.preventDefault();
  const anchor = document.getElementById(styles[`nav-toggle`])!;
  const navBar = document.getElementById('navbar')!;
  anchor.classList.toggle(styles.active);
  navBar.classList.toggle(styles.expand);
};
export default function Navigation() {
  return (
    <header>
      <div className={`${styles.wrap}`}>
        <div className={`${styles.logo}`}>
          <a href="#!">Web Logo</a>
        </div>
        <div className={styles[`nav-mobile`]}>
          <a id={styles[`nav-toggle`]} href="#!" onClick={menuButtonHandler}>
            <span></span>
          </a>
        </div>
        <nav id="navbar" className={styles[`nav-container`]}>
          <ul>
            <li>
              <NavLink to="/" className={activeStyle}>
                Home
              </NavLink>
            </li>
            <li>
              <NavLink to="/new-user" className={activeStyle}>
                Add User
              </NavLink>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
}
