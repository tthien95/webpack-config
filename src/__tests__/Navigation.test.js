import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../components/Navigation/Navigation';

describe('Navigation', () => {
  afterAll(cleanup);

  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    expect(asFragment(<Navigation />)).toMatchSnapshot();
  });
});
