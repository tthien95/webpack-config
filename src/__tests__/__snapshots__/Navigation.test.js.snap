// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="wrap"
    >
      <div
        class="logo"
      >
        <a
          href="#!"
        >
          Web Logo
        </a>
      </div>
      <div
        class="nav-mobile"
      >
        <a
          href="#!"
          id="nav-toggle"
        >
          <span />
        </a>
      </div>
      <nav
        class="nav-container"
        id="navbar"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="nav-active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/new-user"
            >
              Add User
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
